include/mgp_mock.py

```diff
@@ -0,0 +1,1916 @@
+        if not isinstance(messages, _mgp_mock.Messages):
+            raise TypeError("Expected '_mgp_mock.Messages', got '{}'".format(type(messages)))
+        self._messages = messages
+
+    def __deepcopy__(self, memo):
+        # In line with the Python API, this is the same as the shallow copy.
+        return Messages(self._messages)
+
+    def is_valid(self) -> bool:
+        """
+        Check if the `Messages` instance is valid, i.e. if it may be used.
+
+        Returns:
+            A `bool` value that represents whether the instance is in a valid context.
+
+        Examples:
+            ```messages.is_valid()```
+        """
+        return self._messages.is_valid()
+
+    def message_at(self, id: int) -> Message:
+        """
+        Returns the message at the given position.
+
+        Raises:
+            InvalidMessagesError: If the `Messages` instance is not in a valid context.
+        """
+        if not self.is_valid():
+            raise InvalidMessagesError()
+
+        return Message(self._messages.messages[id])
+
+    def total_messages(self) -> int:
+        """
+        Returns how many messages are stored in the `Messages` instance.
+
+        Raises:
+            InvalidMessagesError: If the `Messages` instance is not in a valid context.
+        """
+        if not self.is_valid():
+            raise InvalidMessagesError()
+
+        return len(self._messages.messages)
+
+
+class TransCtx:
+    """The context of the transformation being executed.
+
+    Access to a `TransCtx` is only valid during a single execution of a transformation.
+    You should not globally store a `TransCtx` instance.
+    """
+
+    __slots__ = ("_graph",)
+
+    def __init__(self, graph):
+        if not isinstance(graph, _mgp_mock.Graph):
+            raise TypeError(f"Expected '_mgp_mock.Graph', got '{type(graph)}'")
+
+        self._graph = Graph(graph)
+
+    def is_valid(self) -> bool:
+        """
+        Check if the context is valid, i.e. if the contained structures may be used.
+
+        Returns:
+            A `bool` value that represents whether the context is valid.
+
+        Examples:
+            ```context.is_valid()```
+        """
+        return self._graph.is_valid()
+
+    @property
+    def graph(self) -> Graph:
+        """
+        Access the graph.
+
+        Returns:
+            A `Graph` object representing the graph.
+
+        Raises:
+            InvalidContextError: If the transformation context is not valid.
+
+        Examples:
+            ```context.graph```
+        """
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return self._graph
+
+
+def transformation(func: typing.Callable[..., Record]):
+    raise_if_does_not_meet_requirements(func)
+
+    sig = inspect.signature(func)
+
+    params = tuple(sig.parameters.values())
+    if not params or not params[0].annotation is Messages:
+        if not len(params) == 2 or not params[1].annotation is Messages:
```

Comment:
isinstance I think is the preferred way again

Reply:
Done